storage/remote/max_gauge.go

```diff
@@ -20,7 +20,7 @@ import (
 )
 
 type maxGauge struct {
-	mtx   sync.Mutex
```

Comment:
A normal mutex is fine. RWMutexes are only suitable when there's many many more reads than writes.